x/dex/module.go

```diff
@@ -346,6 +347,12 @@ func (am AppModule) endBlockForContract(ctx sdk.Context, contract types.Contract
 		assetDenomStr := pair.AssetDenom
 		allExistingBuys := am.keeper.GetAllLongBookForPair(ctx, contractAddr, priceDenomStr, assetDenomStr)
 		allExistingSells := am.keeper.GetAllShortBookForPair(ctx, contractAddr, priceDenomStr, assetDenomStr)
+		sort.Slice(allExistingBuys, func(i, j int) bool {
+			return allExistingBuys[i].GetPrice().LT(allExistingBuys[j].GetPrice())
+		})
+		sort.Slice(allExistingSells, func(i, j int) bool {
+			return allExistingSells[i].GetPrice().LT(allExistingSells[j].GetPrice())
```

Comment:
just to confirm, both bid and ask book should be in ascending order correct

Reply:
yeah they should both be in ascending order